daily_milestone.go

```diff
@@ -1,29 +1,65 @@
 package main
 
-import "flag"
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
+var logger *log.Logger
+
+//LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+func getProjectID(BaseURL string, Token string, Projectname string, Namespace string) string {
+	urls := "https://" + BaseURL + "/projects"
 	page := 1
+	strPage := strconv.Itoa(page)
+	s := []string{urls, "?page=", strPage}
+	completeURL := strings.Join(s, "")
 	for {
-
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", completeURL, nil)
+		if err != nil {
+			fmt.Println(err)
+			break
+		}
+		req.Header.Add("Token", "text/plain")
+		req.Header.Add("User-Agent", "SampleClient/1.0")
+		resp, err := client.Do(req)
+		if err != nil {
+			fmt.Print(err)
+			break
+		}
+		defer resp.Body.Close()
+		io.Copy(os.Stdout, resp.Body)
```

Comment:
Not sure if io.Copy is really needed. I think there a more high level function for this in the standard library.

Reply:
ok will look into that